code/modules/antagonists/heretic/items/corrupted_organs.dm

```diff
@@ -0,0 +1,239 @@
+		return
+	var/datum/reagents/extra_reagents = new()
+	extra_reagents.add_reagent(pick(extra_ingredients), amount_added)
+	extra_reagents.trans_to(source, amount_added, transferred_by = src, methods = INJECT)
+
+
+/// Rapidly become hungry if you are not digesting blood
+/obj/item/organ/internal/stomach/corrupt
+	name = "corrupt stomach"
+	desc = "This parasite demands an unwholesome diet in order to be satisfied."
+	/// Do we have an unholy thirst?
+	var/thirst_satiated = FALSE
+	/// Timer for when we get thirsty again
+	var/thirst_timer
+	/// How long until we prompt the player to drink blood again?
+	COOLDOWN_DECLARE(message_cooldown)
+
+/obj/item/organ/internal/stomach/corrupt/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/corrupted_organ)
+	AddElement(/datum/element/noticable_organ, "%PRONOUN_They %PRONOUN_have an unhealthy pallor.")
+
+/obj/item/organ/internal/stomach/corrupt/on_mob_insert(mob/living/carbon/organ_owner, special)
+	. = ..()
+	RegisterSignal(organ_owner, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_drank))
+
+/obj/item/organ/internal/stomach/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	UnregisterSignal(organ_owner, COMSIG_ATOM_EXPOSE_REAGENTS)
+
+/// Check if we drank a little blood
+/obj/item/organ/internal/stomach/corrupt/proc/on_drank(atom/source, list/reagents, datum/reagents/source_reagents, methods)
+	SIGNAL_HANDLER
+	if (!(methods & INGEST))
+		return
+
+	var/contains_blood = locate(/datum/reagent/blood) in reagents
+	if (!contains_blood)
+		return
+
+	if (!thirst_satiated)
+		to_chat(source, span_cult_italic("The thirst is satisfied... for now."))
+	thirst_satiated = TRUE
+	deltimer(thirst_timer)
+	thirst_timer = addtimer(VARSET_CALLBACK(src, thirst_satiated, FALSE), 3 MINUTES, TIMER_STOPPABLE | TIMER_DELETE_ME)
+
+/obj/item/organ/internal/stomach/corrupt/handle_hunger(mob/living/carbon/human/human, seconds_per_tick, times_fired)
+	if (thirst_satiated || human.has_reagent(/datum/reagent/water/holywater))
+		return ..()
+
+	human.adjust_nutrition(-1 * seconds_per_tick)
+
+	if (!COOLDOWN_FINISHED(src, message_cooldown))
+		return ..()
+	COOLDOWN_START(src, message_cooldown, 30 SECONDS)
+
+	var/static/list/blood_messages = list(
+		"Blood...",
+		"Everyone suddenly looks so tasty.",
+		"The blood...",
+		"There's an emptiness in you that only blood can fill.",
+		"You could really go for some blood right now.",
+		"You feel the blood rushing through your veins.",
+		"You think about biting someone's throat.",
+		"Your stomach growls and you feel a metallic taste in your mouth.",
+	)
+	to_chat(human, span_cult_italic(pick(blood_messages)))
+
+	return ..()
+
+
+/// Occasionally bombards you with spooky hands and lets everyone hear your pulse.
+/obj/item/organ/internal/heart/corrupt
+	name = "corrupt heart"
+	desc = "What corruption is this spreading along with the blood?"
+	/// How likely are we to spawn a hand on any particular second?
+	var/hand_chance = 33
```

Comment:
This chance is huge!!! I'd just suicide if I had to dodge a hand every 3 seconds

Reply:
I'll be honest I don't know how these percentages actually work because what with the tick rate it definitely isnt one every three seconds.
I think this one will be more easily tuned and comprehensible if I change it to just use a somewhat randomised cooldown.